src/Analysis/GWASV2/Shared/CohortsOverlapDiagram.jsx

```diff
@@ -0,0 +1,157 @@
+      queryKey: [
+        'checkoverlap',
+        sourceId,
+        selectedStudyPopulationCohort,
+        selectedControlCohort,
+        selectedCovariates,
+        selectedDichotomousCovariates,
+      ],
+      queryFn: () => fetchSimpleOverlapInfo(
+        sourceId,
+        selectedStudyPopulationCohort.cohort_definition_id,
+        selectedControlCohort.cohort_definition_id,
+        selectedCovariates,
+        selectedDichotomousCovariates,
+      ),
+      ...queryConfig,
+    },
+    {
+      queryKey: [
+        'checkoverlap',
+        sourceId,
+        selectedCaseCohort,
+        selectedControlCohort,
+        selectedCovariates,
+        selectedDichotomousCovariates,
+      ],
+      queryFn: () => fetchSimpleOverlapInfo(
+        sourceId,
+        selectedCaseCohort.cohort_definition_id,
+        selectedControlCohort.cohort_definition_id,
+        selectedCovariates,
+        selectedDichotomousCovariates,
+      ),
+      ...queryConfig,
+    },
+    // special case: the overlap of study population with case, excluding any intersection w/ cohort:
+    {
+      queryKey: [
+        'checkoverlap',
+        sourceId,
+        selectedStudyPopulationCohort,
+        selectedCaseCohort,
+        selectedControlCohort,
+        selectedCovariates,
+        selectedDichotomousCovariates,
+      ],
+      queryFn: () => fetchSimpleOverlapInfo(
+        sourceId,
+        selectedStudyPopulationCohort.cohort_definition_id,
+        selectedCaseCohort.cohort_definition_id,
+        selectedCovariates,
+        addCDFilter(selectedCaseCohort.cohort_definition_id,
+          selectedControlCohort.cohort_definition_id,
+          selectedDichotomousCovariates), // ==> adds case/control as extra variable
+      ),
+      ...queryConfig,
+    },
+  ]);
+
+  const {
+    statusPopCase, statusPopControl, statusCaseControl, statusPopCaseControl,
+    dataPopCase, dataPopControl, dataCaseControl, dataPopCaseMinusPopCaseControl
+  } = {
+    statusPopCase: results[0].status,
+    statusPopControl: results[1].status,
+    statusCaseControl: results[2].status,
+    statusPopCaseControl: results[3].status,
+    dataPopCase: results[0].data,
+    dataPopControl: results[1].data,
+    dataCaseControl: results[2].data,
+    dataPopCaseMinusPopCaseControl: results[3].data,
+  };
+
+  // returns 'error' if any error is found, otherwise
+  // returns 'loading' if any result is still loading,
+  // or 'done' if none are loading anymore.
+  const statusResults = (results) => {
+    let status = 'done';
+    for (let i in results) {
+      if (results[i].status === 'error') {
+        return results[i].status;
+      } else if (results[i].status === 'loading') {
+        status = results[i].status;
+      }
+    }
+    return status;
+  }
+
+  if (statusResults(results) === 'loading') {
+    return <Spin />;
+  } else if (statusResults(results) === 'error') {
+    return <React.Fragment>Error getting data for diagram</React.Fragment>;
+  } else {
```

Comment:
ex. `if ([statusPopCase, statusPopControl, statusCaseControl, statusPopCaseControl].some((status) => status === "loading")) {
  return <Spin />
}`

Reply:
Nice. I'll use that 👍